src/Sylius/Bundle/ApiBundle/DataProvider/ProductItemDataProvider.php

```diff
@@ -43,19 +43,19 @@ public function getItem(string $resourceClass, $id, string $operationName = null
     {
         $user = $this->userContext->getUser();
 
+        /** @var ChannelInterface $channel */
+        $channel = $context[ContextKeys::CHANNEL];
+
         if ($user instanceof AdminUserInterface && in_array('ROLE_API_ACCESS', $user->getRoles(), true)) {
             /** @var Product $product */
             $product = $this->productRepository->findOneByCode($id);
```

Comment:
What's the principal different with method `findOneByChannelAndCode()`

Reply:
Honestly, I just modified this class so I thought maybe there's reason behind using different methods,  as one of the purposes of channels is to sell to a specific group of customers

Reply:
Okey, thanks

Reply:
logically, we check whether `$user` is `inctanceof AdminUserInterface` and whether it has access rights to this `$item` further, if any, we find the product and prepare it in order to fill in custom properties and return it as a result of the `getItem` method if `if condition` is not successful we get into the `else` block and do the same there

-------

I think you should check how it was resolved by its vendor and then do your changes because it seems like security issue